app/src/main/java/org/mozilla/fenix/exceptions/ExceptionsAdapter.kt

```diff
@@ -6,49 +6,43 @@ package org.mozilla.fenix.exceptions
 
 import android.view.LayoutInflater
 import android.view.ViewGroup
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.ListAdapter
 import androidx.recyclerview.widget.RecyclerView
 import org.mozilla.fenix.exceptions.viewholders.ExceptionsDeleteButtonViewHolder
 import org.mozilla.fenix.exceptions.viewholders.ExceptionsHeaderViewHolder
 import org.mozilla.fenix.exceptions.viewholders.ExceptionsListItemViewHolder
 
-private sealed class AdapterItem {
+sealed class AdapterItem {
     object DeleteButton : AdapterItem()
     object Header : AdapterItem()
     data class Item(val item: ExceptionsItem) : AdapterItem()
 }
 
-private class ExceptionsList(val exceptions: List<ExceptionsItem>) {
-    val items: List<AdapterItem>
-
-    init {
-        val items = mutableListOf<AdapterItem>()
-        items.add(AdapterItem.Header)
-        for (exception in exceptions) {
-            items.add(AdapterItem.Item(exception))
-        }
-        items.add(AdapterItem.DeleteButton)
-        this.items = items
-    }
-}
-
+/**
+ * Adapter for a list of sites that are exempted from Tracking Protection,
+ * along with controls to remove the exception.
+ */
 class ExceptionsAdapter(
     private val interactor: ExceptionsInteractor
-) : RecyclerView.Adapter<RecyclerView.ViewHolder>() {
-    private var exceptionsList: ExceptionsList = ExceptionsList(emptyList())
+) : ListAdapter<AdapterItem, RecyclerView.ViewHolder>(DiffCallback) {
```

Comment:
Would this change cause a performance issue when the list is large?

Reply:
No, `ListAdapter` does diffing on a background thread so it's more efficient than `notifyDataSetChanged`. 

It might not be a huge win in this case because the list isn't frequently updated, but it helps get rid of code to manage the list.

Reply:
Understood.  Thanks!